### PR TITLE
Fix Username and Password trino and added options

### LINKED ISSUE #1317: Trino ingestion does not build URL correctly when username/password is used
https://github.com/open-metadata/OpenMetadata/blob/main/ingestion/src/metadata/ingestion/source/trino.py#L35

This should append a "@" to the URL.

### PR DESCRIPTION
**Pull request by ayush-shah**
### Describe your changes :

Trino fix #1317 

#
### Type of change :

- [x] Bug fix

#
### Frontend Preview (Screenshots) :
<p align="center">For frontend related change, please link screenshots of your changes preview! Optional for backend related changes.
</p>

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/open-source-community/developer) document.
- [x] I have performed a self-review of my own. 
- [x] I have tagged my reviewers below.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#
### Reviewers

 Ingestion: @harshach

### CODE REVIEW

File: ingestion/src/metadata/ingestion/source/trino.py [hunk 1]
```diff
@@ -28,15 +28,18 @@ class TrinoConfig(SQLConnectionConfig):
...
             url += f"/{quote_plus(self.catalog)}"
             if self.schema_name:
                 url += f"/{quote_plus(self.schema_name)}"
+        if self.username:
+            url += f"?user={quote_plus(self.username)}"
+            if self.password:
+                url += f"&password={quote_plus(self.password)}"
+        if self.options:
+            url = self.get_options(options=self.options, url=url)
+        print(url)
```
Comment (harshach) [inline ingestion/src/metadata/ingestion/source/trino.py:42]: can you remove this print

File: ingestion/src/metadata/ingestion/source/sql_source.py [hunk 2]
```diff
@@ -102,7 +102,19 @@ def get_connection_url(self):
...
         if self.database:
             url += f"/{self.database}"
-        logger.info(url)
+        if self.options:
+            url = self.get_options(options=self.options, url=url)
+
+        logger.debug(url)
+        return url
+
+    def get_options(self, options, url):
```
Comment (harshach) [inline ingestion/src/metadata/ingestion/source/sql_source.py:111]: are we capturing options in SQLSourceConfig, i.e user should be able to pass options via pipeline config in json

Comment (ayush-shah) [reply ingestion/src/metadata/ingestion/source/sql_source.py:111]: yes, User will be able to pass the options in dictionary format via config files

File: ingestion/src/metadata/ingestion/source/trino.py [hunk 3]
```diff
@@ -27,16 +27,17 @@ class TrinoConfig(SQLConnectionConfig):
...
         if self.catalog:
             url += f"/{quote_plus(self.catalog)}"
             if self.schema_name:
                 url += f"/{quote_plus(self.schema_name)}"
+        if self.username:
+            url += f"?user={quote_plus(self.username)}"
+            if self.password:
+                url += f"&password={quote_plus(self.password)}"
+        if self.options:
+            url = self.get_options(options=self.options, url=url)
```
Comment (abrisan) [inline ingestion/src/metadata/ingestion/source/trino.py:40]: May I suggest using [urlencode](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlencode) instead here? It is more idiomatic python code.

File: ingestion/src/metadata/ingestion/source/sql_source.py [hunk 4]
```diff
@@ -102,7 +102,19 @@ def get_connection_url(self):
...
+        if options:
+            url += "?" if "?" not in url else ""
+            for key, value in options.items():
+                if url[-1] != "?":
+                    url += "&"
+                url += f"{key}={value}"
```
Comment (abrisan) [inline ingestion/src/metadata/ingestion/source/sql_source.py:117]: Ditto as below for using [urlencode](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlencode)

File: ingestion/src/metadata/ingestion/source/trino.py (hunk 3 above)
Comment (harshach) [reply ingestion/src/metadata/ingestion/source/trino.py:40]: I think we rant into issues with url encode but we can give it a try

**Pull request closed without merging**